templates/ccng-config.lib.yml

```diff
@@ -1,5 +1,5 @@
 #@ load("@ytt:data","data")
-#@ def ccng_config():
+#@ def/end ccng_config():
```

Comment:
I dunno enough `ytt` to understand this well. What does `def/end` do differently here vs having the `end` later?

Reply:
Short version is, this way you can't accidentally try to add some YAML after the "end," which will cause `ytt` to throw a very perplexing error message.

Longer version is, `/end` ensures that the `end` will be applied at the end of the next... not sure what it's called. Unit of evaluation? In this case it's the end of the document, but in other cases it would be the next line. If there's dangling YAML present after the `end` statement, you'll trigger this `ytt` bug, where it generates an invalid Starlark program. https://github.com/k14s/ytt/issues/225

Reply:
Thanks for that context!

Reply:
FWIW: a `ytt` annotation applies to the node directly below it.

Here, the
```
#@ if/end cc_config():
---
```
applies to the Document node (denoted by the YAML document start: `---`).

A Document node is a root node, holding all the fields and arrays underneath it.
So, this `/end` ensures that the remainder of the file is part of the function enclosing the YAML fragment.

It might be helpful is to point out that `#@ ` (note the space after the `@`) is short-hand for the annotation `#@template/code` (described in [Reference > Templating > Annotation Format](https://github.com/k14s/ytt/blob/develop/docs/lang-ref-annotation.md#shared-templating-annotations)). So even when you see what looks like "just code" is, in fact, annotated in.